taskcluster/ci/module-build/kind.yml

```diff
@@ -27,16 +25,25 @@ job-defaults:
   worker:
     docker-image: { in-tree: linux }
     max-run-time: 1800
-    script: |
-      rsync -a /builds/worker/fetches/libs/ /builds/worker/checkouts/src/libs/
-      echo "rust.targets=arm,arm64,x86_64,x86,darwin,linux-x86-64,win32-x86-64-gnu\n" > local.properties
-      ./gradlew --no-daemon ":{module_name}:testDebug"
-      ./gradlew --no-daemon ":{module_name}:assembleRelease"
-      ./gradlew --no-daemon ":{module_name}:publish"
-      ./gradlew --no-daemon ":{module_name}:checkMavenArtifacts"
   run:
-    using: run-task
-    cwd: "{checkout}"
+    pre-gradlew:
+      - [taskcluster/scripts/toolchain/rustup-setup.sh]
+      # FIXME: scripts subshell at runtime so we need to source this here
+      # to be able to access `rustup` and `rustc` from within the cc script
+      # and the gradle command. Another options could be to set those env vars
+      # here like: [export, 'PATH=$HOME/.cargo/bin:$PATH'
+      - [source, taskcluster/scripts/toolchain/rustup-setup.sh]
+      - [taskcluster/scripts/toolchain/cross-compile-setup.sh]
+      - [rsync, '-a', /builds/worker/fetches/libs/, /builds/worker/checkouts/src/libs/]
+      - [echo, 'rust.targets=arm,arm64,x86_64,x86,darwin,linux-x86-64,win32-x86-64-gnu\n', '>', local.properties]
+    gradlew:
+      - ':{module_name}:testDebug'
```

Comment:
Unclear to me whether chaining the `gradlew`commands is the same as running them separately, one after another.

Reply:
(Drive-by comment) It should be semantically the same to have them each on their own, but it will most likely be a bit slower, especially if `--no-daemon` is involved, as it parses the gradle task configuration again and again on each invocation

Reply:
Yeah, you're totally right. My follow-up question would've been how come it now takes 9m40s for a build that use to take combined around 5mins. I can re-work this part and rely on `run-commands` instead so that I can use the `--no-daemon` under the skin. CC: @JohanLorenzo

Reply:
> Unclear to me whether chaining the gradlewcommands is the same as running them separately, one after another.

It'll be in one big `gradlew` command. So gradle will be in charge of scheduling the each step the way it wants. Like @badboy said, it's faster. In some circumstances, you may want to break the `gradlew` command into several pieces. For instance: you want to be sure that one command is finished before the next one starts. That said, in the general case you want them all in one big command.

When you have one big command, `--no-deamon` is the way to go, I think. This way, you don't have to spin a deamon that will be used once. If you have several commands, then you likely want to start the daemon.